openml/setups/functions.py

```diff
@@ -75,37 +89,39 @@ def initialize_model(setup_id):
         model : sklearn model
             the scikitlearn model with all parameters initailized
     '''
-    def _to_dict_of_dicts(_params):
-        # this subfunction transforms an openml setup object into
-        # a dict of dicts, structured: flow_id maps to dict of
-        # parameter_names mapping to parameter_value
-        _res = {}
-        for _param in _params:
-            _flow_id = _params[_param].flow_id
-            _param_name = _params[_param].parameter_name
-            _param_value = _params[_param].value
-            if _flow_id not in _res:
-                _res[_flow_id] = {}
-            _res[_flow_id][_param_name] = _param_value
-        return _res
+
+    # transform an openml setup object into
+    # a dict of dicts, structured: flow_id maps to dict of
+    # parameter_names mapping to parameter_value
+
+    setup = get_setup(setup_id)
+    parameters = {}
+    for _param in setup.parameters:
+        _flow_id = setup.parameters[_param].flow_id
+        _param_name = setup.parameters[_param].parameter_name
+        _param_value = setup.parameters[_param].value
+        if _flow_id not in parameters:
+            parameters[_flow_id] = {}
+        parameters[_flow_id][_param_name] = _param_value
 
     def _reconstruct_flow(_flow, _params):
-        # sets the values of flow parameters (and subflows) to
```

Comment:
small todo (sorry, should have been me in prev pull request): document what types _flow (flow object?) and _params (totally forgot) are? this would make this function better understandable